tests/test_reader/test_demultiplexing_parsers.py

```diff
@@ -8,33 +8,59 @@
     def test(self):
-        conversion_stat = os.path.join(self.assets_path, 'test_crawlers', 'ConversionStats.xml')
+        # testing multiplexed run stats parsing (previously from ConversionStats.xml and AdapterTrimming.txt')
+        json_stat = os.path.join(self.assets_path, 'test_crawlers', 'Stats.json')
+        with open(json_stat, 'r') as stats:
+            json_data = json.load(stats)
+
         expected_barcode_per_lane = [
-            ('default', 'Undetermined', '1', 'unknown', 1696088, 80740, 12111000, 9937085, 7746149),
-            ('default', 'Undetermined', '2', 'unknown', 2335276, 122074, 18311100, 15157909, 12445247),
-            ('10015AT', 'LP6002014-DTP_A01', '1', 'ATTACTCG', 537099, 537099, 80564850, 72789430, 58579087),
-            ('10015AT', 'LP6002014-DTP_A01', '2', 'ATTACTCG', 466412, 466412, 69961800, 61852875, 49303565),
-            ('10015AT', 'LP6002014-DTP_A02', '1', 'TCCGGAGA', 539999, 539999, 80999850, 73257234, 57184580),
-            ('10015AT', 'LP6002014-DTP_A02', '2', 'TCCGGAGA', 469731, 469731, 70459650, 62355216, 48408701)
+            ('LP6002014-DTP_A01', '1', 'ATTACTCG', 537099, 537099, 80564850, 72789430, 58579087),
+            ('LP6002014-DTP_A02', '1', 'TCCGGAGA', 539999, 539999, 80999850, 73257234, 57184580),
+            ('Undetermined', '1', 'unknown', 1696088, 80740, 12111000, 9937085, 7746149),
+            ('LP6002014-DTP_A01', '2', 'ATTACTCG', 466412, 466412, 69961800, 61852875, 49303565),
+            ('LP6002014-DTP_A02', '2', 'TCCGGAGA', 469731, 469731, 70459650, 62355216, 48408701),
+            ('Undetermined', '2', 'unknown', 2335276, 122074, 18311100, 15157909, 12445247)
         ]
         expected_unknown_barcodes_per_lanes = [
-            ('1', 'CCCCCCCC', '4695'),
-            ('1', 'TGAAGCTA', '828'),
-            ('2', 'CCCCCCCC', '22153'),
-            ('2', 'NCCCCCCC', '9133')
+            ('1', 'CCCCCCCC', '4820'),
+            ('1', 'TGAAGCTA', '660'),
+            ('2', 'CCCCCCCC', '21320'),
+            ('2', 'NCCCCCCC', '9840')
+        ]
+        barcodes, unknowns, adapter_trimmed_by_id = dm.parse_json_stats(json_data, 'test_run_id')
+        self.assertCountEqual(barcodes,expected_barcode_per_lane)
+        self.assertCountEqual(unknowns,expected_unknown_barcodes_per_lanes)
+
+        self.assertCountEqual(adapter_trimmed_by_id, {
+            'test_run_id_1_ATTACTCG': {'read_1_trimmed_bases': 1056218, 'read_2_trimmed_bases': 1080753},
+            'test_run_id_1_TCCGGAGA': {'read_1_trimmed_bases': 974374, 'read_2_trimmed_bases': 992558},
+            'test_run_id_1_unknown': {'read_1_trimmed_bases': 100051, 'read_2_trimmed_bases': 122287},
+            'test_run_id_2_ATTACTCG': {'read_1_trimmed_bases': 866438, 'read_2_trimmed_bases': 914963},
+            'test_run_id_2_TCCGGAGA': {'read_1_trimmed_bases': 797469, 'read_2_trimmed_bases': 843785},
+            'test_run_id_2_unknown': {'read_1_trimmed_bases': 112371, 'read_2_trimmed_bases': 143299}
```

Comment:
`assertCountEqual` doesn't pick up differences in dict values (it casts the observed and expected arguments to lists, giving just the dict keys), so the build still passes if the numbers are changed. `assertEqual` would be a better check here

Reply:
@mwhamgenomics I was using assertEqual before I used assertCountEqual, but it turns out assertEqual checks the order of the elements in the dict too, which was not always deterministic. I'll try to find a better solution.